consensus/hotstuff/cruisectl/block_time_controller.go

```diff
@@ -462,16 +448,35 @@ func (ctl *BlockTimeController) measureViewDuration(tb TimedBlock) error {
 	return nil
 }
 
-// processEpochSetupPhaseStarted processes EpochSetupPhaseStarted events from the protocol state.
+// processEpochExtended processes the EpochExtended protocol data.
+// Whenever we encounter an epoch extension, we:
+//   - set epochFallbackTriggered to true if this is the first extension encountered.
+//   - update the curr epoch final view and target end time with the extension data.
+//
+// No errors are expected during normal operation.
+func (ctl *BlockTimeController) processEpochExtended(_ uint64, _ *flow.Header, extension flow.EpochExtension) error {
+	if !ctl.epochFallbackTriggered {
+		// set epochFallbackTriggered to true when we encounter the first extension
+		ctl.epochFallbackTriggered = true
+	}
+	ctl.curEpochFinalView = extension.FinalView
+	ctl.curEpochTargetEndTime = extension.TargetEndTime
```

Comment:
Just spoke with @durkmurder about this as he's planning to pick up #6194. We agreed that removing `TargetEndTime` from the extension model and moving that logic into the `protocol.Epoch` API method makes sense (reduces some duplication). That would require us to query `state.AtBlockID(refBlock).Epochs().Current().TargetEndTime()` here, rather than reading directly from the extension.

Reply:
Sounds good

Reply:
https://github.com/onflow/flow-go/pull/6102/commits/1f605be6ed39d809328e69e789eb489f79dbd538